rewrite-java/src/main/java/org/openrewrite/java/cleanup/NoDoubleBraceInitialization.java

```diff
@@ -207,4 +223,53 @@ public J.Block visitBlock(J.Block block, ExecutionContext ctx) {
             return bl;
         }
     }
+
+    private static class AddWarningMessage extends JavaIsoVisitor<String> {
+        private int count = 0;
+
+        static J.NewClass addWarningComment(J.NewClass nc, String methodName) {
+            return (J.NewClass) new AddWarningMessage().visit(nc, methodName);
+        }
+
+        @Override
+        public J.NewClass visitNewClass(J.NewClass newClass, String methodName) {
+            count++;
+            // skip the root which starts from 1, and only comment for the 1st raw constructor which has count as 2.
+            if (count == 2) {
+                String comment = "Did you mean to invoke " + methodName + " method to the collection?";
+                return SearchResult.found(newClass, comment);
+            }
+
+            return super.visitNewClass(newClass, methodName);
+        }
+    }
+
+    private static class FindMethodInvocationInDoubleBrace extends JavaIsoVisitor<AtomicBoolean> {
+        /**
+         * Find whether any collection content initialization method(e.g add() or put()) is invoked in the double brace.
+         * @param j The subtree to search, supposed to be the 2nd brace (J.Block)
+         * @return true if any method invocation found in the double brace, otherwise false.
+         */
+        static AtomicBoolean find(J j) {
+            return new FindMethodInvocationInDoubleBrace()
+                .reduce(j, new AtomicBoolean());
+        }
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, AtomicBoolean atomicBoolean) {
+            if (atomicBoolean.get()) {
+                return method;
+            }
+
+            // In double brace initialization of a new class, methods to initialize collection content like `add()` or
+            // `put()` are all belonged to the anonymous class and have type `JavaType.Parameterized`.
+            if (method.getMethodType() != null
+                && method.getMethodType().getDeclaringType() instanceof JavaType.Parameterized) {
```

Comment:
@kunli2 The containing class could of course also be a parameterized type. It seems like it should be possible to check that the invocation's `select` (i.e. receiver) is either `null`, `super`, or `this`. In case of `null` it would probably be possible to further check if the method is declared by a type in the anonymous subclass type hierarchy. What do you think?